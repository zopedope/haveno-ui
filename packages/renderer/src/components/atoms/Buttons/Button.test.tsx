// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { Button, TextButton } from ".";

describe("atoms::Buttons", () => {
  it("renders primary button by default", () => {
    const { asFragment } = render(<Button>Primary</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders neutral button", () => {
    const { asFragment } = render(<Button flavor="neutral">Neutral</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders success button", () => {
    const { asFragment } = render(<Button flavor="success">Success</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders error button", () => {
    const { asFragment } = render(<Button flavor="danger">Error</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders text button", () => {
    const { asFragment } = render(
      <TextButton>This is a Text Button</TextButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
