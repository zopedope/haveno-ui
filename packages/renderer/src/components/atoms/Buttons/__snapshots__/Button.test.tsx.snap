// Vitest Snapshot v1

exports[`atoms::Buttons > renders error button 1`] = `
<DocumentFragment>
  <button
    class="mantine-Button-filled mantine-Button-root mantine-b4sl6"
    type="button"
  >
    <div
      class="mantine-3xbgk5 mantine-Button-inner"
    >
      <span
        class="mantine-qo1k2 mantine-Button-label"
      >
        Error
      </span>
    </div>
  </button>
</DocumentFragment>
`;

exports[`atoms::Buttons > renders neutral button 1`] = `
<DocumentFragment>
  <button
    class="mantine-Button-filled mantine-Button-root mantine-h60jl4"
    type="button"
  >
    <div
      class="mantine-3xbgk5 mantine-Button-inner"
    >
      <span
        class="mantine-qo1k2 mantine-Button-label"
      >
        Neutral
      </span>
    </div>
  </button>
</DocumentFragment>
`;

exports[`atoms::Buttons > renders primary button by default 1`] = `
<DocumentFragment>
  <button
    class="mantine-Button-filled mantine-Button-root mantine-11hawrf"
    type="button"
  >
    <div
      class="mantine-3xbgk5 mantine-Button-inner"
    >
      <span
        class="mantine-qo1k2 mantine-Button-label"
      >
        Primary
      </span>
    </div>
  </button>
</DocumentFragment>
`;

exports[`atoms::Buttons > renders success button 1`] = `
<DocumentFragment>
  <button
    class="mantine-Button-filled mantine-Button-root mantine-1wbcmdw"
    type="button"
  >
    <div
      class="mantine-3xbgk5 mantine-Button-inner"
    >
      <span
        class="mantine-qo1k2 mantine-Button-label"
      >
        Success
      </span>
    </div>
  </button>
</DocumentFragment>
`;

exports[`atoms::Buttons > renders text button 1`] = `
<DocumentFragment>
  <button
    class="mantine-UnstyledButton-root mantine-jxthji"
    type="button"
  >
    <span
      class="mantine-Text-root mantine-1xu8ypy"
    >
      This is a Text Button
    </span>
  </button>
</DocumentFragment>
`;
